packages/internal/src/utils.js

```diff
@@ -381,3 +381,25 @@ export const fsStreamReady = stream =>
     stream.on('ready', onReady);
     stream.on('error', onError);
   });
+
+/**
+ * Concise way to check values are available from object literal shorthand.
+ * Throws error message to specify the missing values.
+ *
+ * @template {Record<string, unknown>} T
+ * @param {T} obj
+ * @throws if any value in the object entries is not defined
+ * @returns {asserts obj is Required<T>}
+ */
+
+export const assertDefined = obj => {
+  const missing = [];
+  for (const [key, val] of Object.entries(obj)) {
+    if (val === undefined) {
+      missing.push(key);
+    }
+  }
+  if (missing.length > 0) {
+    Fail`missing ${missing.join(',')}`;
```

Comment:
I doubt there are any plausible uses of this where you want to redact the contents of missing, since it reveals on the property names and not the values. So

```suggestion
    Fail`missing ${q(missing.join(','))}`;
```
or likely better
```suggestion
    Fail`missing ${q(missing)}`;
```
depending on what `q` does here.

Reply:
done